src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -225,12 +230,7 @@ internal void Validate()
 
         internal bool IsDefaultHubName()
         {
-            return this.isDefaultHubName ?? this.hubName == null;
-        }
-
-        internal void SetDefaultHubName(string defaultHubName)
-        {
-            this.hubName = defaultHubName;
+            return string.Equals(this.defaultHubName, this.hubName);
```

Comment:
Task hub names are case insensitive, so you should use a case-insensitive string comparison here.